shopinvader_stock_inventory/services/stock.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/lgpl)
+
+from datetime import datetime
+
+from odoo import fields
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+
+
+class ProductService(Component):
+    _inherit = ["base.shopinvader.service"]
+    _name = "shopinvader.stock.service"
+    _usage = "stock"
+    _expose_model = "product.product"
+
+    def get(self, _id=None, **params):
+        """Return the product specified by _id."""
+        if _id:
+            return self._to_json_one(self._get(_id))
+        else:
+            return self.search(**params)
+
+    def search(self, **params):
+        """Returns all product or all product in category."""
+        params["domain"] = []
+        if "per_page" not in params:
```

Comment:
this domain should come from the request. If you have special requirements for your project, override it locally.